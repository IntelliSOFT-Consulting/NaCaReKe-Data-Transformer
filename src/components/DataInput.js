import React from 'react';
import { Button, Upload } from 'antd';
import { UploadOutlined } from '@ant-design/icons';

export default function DataInput(props) {
  const config = {
    onChange: info => {
      if (info.fileList && info.fileList.length == 1) {
        props.handleFile(info.fileList[0]?.originFileObj);
      } else if (info.fileList && info.fileList.length > 1) {
        props.handleFile(
          info.fileList[info.fileList.length - 1]?.originFileObj
        );
      }
    },
    showUploadList: false,
  };

  const SheetJSFT = [
    'xlsx',
    'xlsb',
    'xlsm',
    'xls',
    'xml',
    'csv',
    'txt',
    'ods',
    'fods',
    'uos',
    'sylk',
    'dif',
    'dbf',
    'prn',
    'qpw',
    '123',
    'wb*',
    'wq*',
    'html',
    'htm',
    'json'
  ]
    .map(x => '.' + x)
    .join(',');

  return (
    <Upload data-testid='file-upload' {...config} accept={SheetJSFT}>
      <Button data-testid='upload-btn' icon={<UploadOutlined />}>Click to Upload</Button>
    </Upload>
  );
}
