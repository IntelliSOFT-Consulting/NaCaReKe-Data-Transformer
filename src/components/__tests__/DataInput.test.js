/* eslint-disable no-undef */
import DataInput from '../DataInput';

it('Renders the upload field correctly', () => {
  const spy = sinon.spy();
  const input = mount(<DataInput handleFile={spy} />);
  expect(input).toBeTruthy();
  const file = input.find('input[type="file"]');
  expect(file).toBeTruthy();
});

it('Contains upload text', () => {
  const spy = sinon.spy();
  const element = mount(<DataInput handleFile={spy} />);
  const uploadText = element.find('button').text();
  expect(uploadText).toBeTruthy();
});

it('matches snapshot', () => {
  const spy = sinon.spy();
  const tree = shallow(<DataInput handleFile={spy} />);
  expect(tree).toMatchSnapshot();
});
