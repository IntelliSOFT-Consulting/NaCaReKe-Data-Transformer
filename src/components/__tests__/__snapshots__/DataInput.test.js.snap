// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Upload
  accept=".xlsx,.xlsb,.xlsm,.xls,.xml,.csv,.txt,.ods,.fods,.uos,.sylk,.dif,.dbf,.prn,.qpw,.123,.wb*,.wq*,.html,.htm,.json"
  action=""
  className=""
  data={Object {}}
  data-testid="file-upload"
  disabled={false}
  listType="text"
  multiple={false}
  onChange={[Function]}
  showUploadList={false}
  supportServerRender={true}
  type="select"
>
  <Button
    data-testid="upload-btn"
    icon={<ForwardRef(UploadOutlined) />}
  >
    Click to Upload
  </Button>
</Upload>
`;
